coercion.ts

```diff
@@ -241,65 +185,60 @@ export function enableTypeCoercion<
             [T, ...PipeItem<unknown, unknown, BaseIssue<unknown>>[]]
           >),
 ): {
-  coerced: boolean;
-  schema:
-    | ReturnType<typeof coerce>
-    | ReturnType<typeof generateReturnSchema>
-    | (T extends GenericSchema
-        ? SchemaWithPipe<
-            [T, ...PipeItem<unknown, unknown, BaseIssue<unknown>>[]]
-          >
-        : SchemaWithPipeAsync<
-            [T, ...PipeItem<unknown, unknown, BaseIssue<unknown>>[]]
-          >);
+  transformAction: TransformAction<unknown, unknown> | undefined;
+  schema: GenericSchema | GenericSchemaAsync;
 } {
-  const originalSchema = "pipe" in type ? type.pipe[0] : type;
+  if ("pipe" in type) {
+    const { transformAction, schema: coercedSchema } = enableTypeCoercion(
+      type.pipe[0],
+    );
+    const schema = type.async
+      ? pipeAsync(coercedSchema, ...type.pipe.slice(1))
+      : // @ts-expect-error `coercedSchema` must be sync here but TypeScript can't infer that.
+        pipe(coercedSchema, ...type.pipe.slice(1));
```

Comment:
Current code determines whether we should use `pipeAsync` or `pipe` by checking the original schema and the coerced schema: https://github.com/chimame/conform-to-valibot/blob/4e1e57bebd9d5a6780c0d72587a78bfa8a588f83/coercion.ts#L149

However it is enough to check the original schema. Checking both schemas causes issue #55.